src/pgduckdb_types.cpp

```diff
@@ -488,79 +617,102 @@ numeric_typmod_scale(int32 typmod) {
 	return (((typmod - VARHDRSZ) & 0x7ff) ^ 1024) - 1024;
 }
 
-static duckdb::LogicalType
-ChildTypeFromArray(Oid array_type) {
-	switch (array_type) {
-	case CHARARRAYOID:
-		return duckdb::LogicalTypeId::TINYINT;
-	case BOOLARRAYOID:
-		return duckdb::LogicalTypeId::BOOLEAN;
-	case INT4ARRAYOID:
-		return duckdb::LogicalTypeId::INTEGER;
-	case INT8ARRAYOID:
-		return duckdb::LogicalTypeId::BIGINT;
-	default:
-		throw duckdb::NotImplementedException("No child type set for Postgres OID %d", array_type);
-	}
-}
-
 duckdb::LogicalType
 ConvertPostgresToDuckColumnType(Form_pg_attribute &attribute) {
 	auto &type = attribute->atttypid;
 	auto &typmod = attribute->atttypmod;
 	auto dimensions = attribute->attndims;
+	duckdb::LogicalType base_type;
 	switch (type) {
 	case BOOLOID:
-		return duckdb::LogicalTypeId::BOOLEAN;
+	case BOOLARRAYOID: {
+		base_type = duckdb::LogicalTypeId::BOOLEAN;
+		break;
+	}
 	case CHAROID:
-		return duckdb::LogicalTypeId::TINYINT;
+	case CHARARRAYOID: {
+		base_type = duckdb::LogicalTypeId::TINYINT;
+		break;
+	}
 	case INT2OID:
-		return duckdb::LogicalTypeId::SMALLINT;
+	case INT2ARRAYOID: {
+		base_type = duckdb::LogicalTypeId::SMALLINT;
+		break;
+	}
 	case INT4OID:
-		return duckdb::LogicalTypeId::INTEGER;
+	case INT4ARRAYOID: {
+		base_type = duckdb::LogicalTypeId::INTEGER;
+		break;
+	}
 	case INT8OID:
-		return duckdb::LogicalTypeId::BIGINT;
+	case INT8ARRAYOID: {
+		base_type = duckdb::LogicalTypeId::BIGINT;
+		break;
+	}
 	case BPCHAROID:
+	case BPCHARARRAYOID:
 	case TEXTOID:
-	case VARCHAROID:
-		return duckdb::LogicalTypeId::VARCHAR;
+	case TEXTARRAYOID:
+	case VARCHARARRAYOID:
+	case VARCHAROID: {
+		base_type = duckdb::LogicalTypeId::VARCHAR;
+		break;
+	}
 	case DATEOID:
 		return duckdb::LogicalTypeId::DATE;
```

Comment:
This seems to be missing `DATEARRAYOID` by accident.
```suggestion
	case DATEOID:
	case DATEARRAYOID: {
		base_type = duckdb::LogicalTypeId::DATE;
		break;
	}
```